qbft/timeout.go

```diff
@@ -1,18 +1,85 @@
 package qbft
 
 import (
+	"github.com/attestantio/go-eth2-client/spec/phase0"
+	"github.com/bloxapp/ssv-spec/types"
 	"github.com/pkg/errors"
-	"time"
+	"golang.org/x/exp/constraints"
 )
 
 var (
-	quickTimeoutThreshold = Round(8)        //nolint
-	quickTimeout          = 2 * time.Second //nolint
-	slowTimeout           = 2 * time.Minute //nolint
+	// quickTimeoutThreshold is the round after which the timeout duration increases
+	quickTimeoutThreshold = Round(6) //nolint
+	// quickTimeout is the timeout in seconds for the first 6 rounds
+	quickTimeout int64 = 2 // 2 seconds
+	// slowTimeout is the timeout in seconds for rounds after the first 86
+	slowTimeout int64 = 120 // 2 minutes
 	// CutoffRound which round the instance should stop its timer and progress no further
-	CutoffRound = 15 // stop processing instances after 8*2+120*6 = 14.2 min (~ 2 epochs)
+	CutoffRound = 14 // stop processing instances after 6*2+120*7 = 14.2 min (~ 2 epochs)
 )
 
+type RoundTimer struct {
+	// role is the role of the QBFT instance
+	Role types.BeaconRole
+	// height is the current height of the instance
+	Height Height
+	// network is the beacon network
+	Network types.BeaconNetwork
+	//current unix epoch time in seconds
+	CurrentTime int64
```

Comment:
I think we can use `time.Time` and `time.Duration` instead of integers, makes it a bit more friendly.

For tests, you can use `time.Unix()` to create time.Time from an integer.

Reply:
I intentionally don't want the spec to rely on go standard lib